File: VB6/Unibanco/Fontes/Logistica 2002/gestao integrada/prototipo/HTML - Homologado/Scripts/Funcoes.js
<script language="JScript">
Array.prototype.IsArray = "true";
String.prototype.IsArray = "false";

function Menu( Mostra )
{
  if( Mostra )
  { 
    /* Mostrar Menu e Diminuir Página... */
    document.all.DMenuOpen.style.visibility  = "hidden" ; 
    document.all.DMenuClose.style.visibility = "visible" ;
    document.all.BG_UBB.style.left           = -159 ;     
  }
  else
  {
    /* Esconder Menu e Aumentar Página... */
    document.all.DMenuOpen.style.visibility  = "visible" ; 
    document.all.DMenuClose.style.visibility = "hidden" ;
    document.all.BG_UBB.style.left           = 0 ;     
  }
  parent.MenuOn( Mostra );
}

function Inicializa()
{
  var Teste;
  Teste = Comunica("", "");
}


function Comunica(xmldoc, strOpcao)
{    
  var xmlhttp = new ActiveXObject("Msxml2.XMLHTTP");
  xmlhttp.Open("POST", "processa.asp?Opcao=" + strOpcao, false);
  xmlhttp.Send(xmldoc);
  return(xmlhttp.responseText);
}

function MontaArray(objDoc, Nivel)
{
  var arResult, arVariant, intLinhas, intColunas, intCol, intRow, i;
  
  arResult = new Array();
  for (i = 0; i < objDoc.documentElement.childNodes.length; i++)
  {
    if (objDoc.documentElement.childNodes[i].childNodes[0].nodeName == "Variant")
    {
      intLinhas = objDoc.documentElement.childNodes[i].childNodes[0].childNodes.length;
      intColunas = objDoc.documentElement.childNodes[i].childNodes[0].childNodes[0].childNodes.length;
      arVariant = new Array();
      for (intCol = 0; intCol < intColunas; intCol++)
      {
        arVariant[intCol] = new Array();
        for (intRow = 0; intRow < intLinhas; intRow++)
        {
          if (objDoc.documentElement.childNodes[i].childNodes[0].childNodes[intRow].childNodes[intCol].childNodes.length != 0)
          {
            if (objDoc.documentElement.childNodes[i].childNodes[0].childNodes[intRow].childNodes[intCol].childNodes[0].nodeName == "Variant")
            {
              var objDoc2 = new ActiveXObject("MSXML2.DOMDocument");
              objDoc2.loadXML ("<root><parametro>" + objDoc.documentElement.childNodes[i].childNodes[0].childNodes[intRow].childNodes[intCol].childNodes[0].xml + "</parametro></root>");
              arVariant[intCol][intRow] = MontaArray(objDoc2, 1);
            }
            else
            {
              arVariant[intCol][intRow] = objDoc.documentElement.childNodes[i].childNodes[0].childNodes[intRow].childNodes[intCol].text;
            }
          }
          else
          {
            arVariant[intCol][intRow] = objDoc.documentElement.childNodes[i].childNodes[0].childNodes[intRow].childNodes[intCol].text;
          }

        }
      }
      if (Nivel == null)
      {
        arResult[i] = arVariant;
      }
      else
      {
        arResult = arVariant;
      }
    }    
    else
    {
      arResult[i] = objDoc.documentElement.childNodes[i].childNodes[0].text;
    }
  }
return arResult;
}

function MontaXML(vntArray, Nivel)
{
  var intRow, intCol, intPos;
  var strCol, strLinha, strVariant, strInstrucao, vntTeste = new Array();
  var NodePrincipal, NodeParametro, NodeTipo, NodeLinha, NodeColuna;
  
  var objDoc = new ActiveXObject("MSXML2.DOMDocument");

  if (Nivel == null)
  {
    strInstrucao = objDoc.createProcessingInstruction("xml", "version='1.0'");
    strInstrucao = objDoc.appendChild(strInstrucao);

    NodePrincipal = objDoc.createNode(1,"root", "");  
  }


  for (intPos = 0; intPos <= vntArray.length - 1; intPos++)
  {
    if (Nivel == null)
    {
      NodeParametro = objDoc.createNode(1,"Parametro", "");
    }
    
    if (vntArray[intPos].IsArray == "false")
    {
      NodeTipo = objDoc.createNode(1, "String", "");      
      NodeTipo.text = vntArray[intPos];
    }
    else
    {
      NodeTipo = objDoc.createNode(1, "Variant", "");
      for (intRow = 0; intRow <= vntArray[intPos][0].length - 1; intRow++)
      {
        strCol = "";
        NodeLinha = objDoc.createNode(1, "Linha", "");
        for (intCol = 0; intCol <= vntArray[intPos].length - 1; intCol++)
        {
          NodeColuna = objDoc.createNode(1, "Coluna", "");

          if (vntArray[intPos][intCol][intRow].IsArray == "true")
          {
            vntTeste[0] = vntArray[intPos][intCol][intRow];
            NodeColuna.text = MontaXML(vntTeste, 1);
          }
          else
    {
      NodeColuna.text = vntArray[intPos][intCol][intRow];
    }
          
          NodeColuna = NodeLinha.insertBefore(NodeColuna, null);
        }
        NodeLinha = NodeTipo.insertBefore(NodeLinha, null);
      }
    }
    if (Nivel == null)
    {
      NodeTipo = NodeParametro.insertBefore(NodeTipo, null);
      NodeParametro = NodePrincipal.insertBefore(NodeParametro, null);
    }
  }
  if (Nivel == null)
  {
    NodePrincipal = objDoc.insertBefore(NodePrincipal, null);
  }
  else
  {
    NodeTipo = objDoc.insertBefore(NodeTipo, null);
  }

  var cond1 = /&lt;/g;
  var cond2 = /&gt;/g;
  
  return objDoc.xml.replace(cond1, "<").replace(cond2, ">");
}

function IsNum(passedVal)
{
  if (passedVal == "")
  {
    return false;
  }
  for (i = 0; i < passedVal.length; i++)
  {
    if (passedVal.charAt(i) < "0")
    {
      return false;
    }
    if (passedVal.charAt(i) > "9")
        {
          return false;
    }
  }
  return true;
}


function turnOn(imageName)
{
  if (document.images)
  {
    document[imageName].src = eval(imageName + "on.src");
    document.body.style.cursor = "hand"
  }
}

function turnOff(imageName)
{
  if (document.images)
  {
    document[imageName].src = eval(imageName + "off.src");
    document.body.style.cursor = "default"
  }
}

function PreLoadImages()
{
  for (i = 0; i < document.images.length; i++)
  {
    eval(document.images(i).name + "on = new Image()");
    eval(document.images(i).name + "on.src = 'Imagens/" + document.images(i).name + "B.gif'");
    eval(document.images(i).name + "off = new Image()");
    eval(document.images(i).name + "off.src = 'Imagens/" + document.images(i).name + ".gif'");
  }
}

function Mid(String, Start, Length)
{
  if (String == null)
    return (false);

  if (Start > String.length)
    return '';

  if (Length == null || Length.length == 0)
    return (false);

  return String.substr((Start - 1), Length);
}


function IsNumeric(Str)
{
  var i;
  var char;
  
  for( i = 0; i < Str.length ; i++ )
  {
    char = Mid( Str, i, 1);
    if( char != "0" && char != "1"  && char != "2"  && char != "3"  && char != "4"  && 
        char != "5" && char != "6"  && char != "7"  && char != "8"  && char != "9"      )
      return false;
  }
  return true;
}

</script>

File: VB6/Unibanco/Fontes/Logistica 2002/gestao integrada/prototipo/HTML - Homologado/Scripts/Funcoes.vb
<%
Function MontaXML(vntArray, Nivel)
  Dim intRow, intCol, intPos, vntTeste(0)
  Dim strCol, strLinha, strVariant, strInstrucao
  Dim NodePrincipal, NodeParametro, NodeTipo, NodeLinha, NodeColuna
  
  Dim objDoc
  Set objDoc = CreateObject("MSXML2.DOMDocument")

  If Nivel = 0 Then
    Set strInstrucao = objDoc.createProcessingInstruction("xml", "version=""1.0""")
    Set strInstrucao = objDoc.appendChild(strInstrucao)

    Set NodePrincipal = objDoc.createNode(1, "root", "")
  End If

  For intPos = LBound(vntArray) To UBound(vntArray)
    If Nivel = 0 Then Set NodeParametro = objDoc.createNode(1, "Parametro", "")
    If Not IsArray(vntArray(intPos)) Then
      Set NodeTipo = objDoc.createNode(1, "String", "")
      NodeTipo.Text = vntArray(intPos)
    Else
      Set NodeTipo = objDoc.createNode(1, "Variant", "")
      For intRow = LBound(vntArray(intPos), 2) To UBound(vntArray(intPos), 2)
        strCol = ""
        Set NodeLinha = objDoc.createNode(1, "Linha", "")
        For intCol = LBound(vntArray(intPos), 1) To UBound(vntArray(intPos), 1)
          Set NodeColuna = objDoc.createNode(1, "Coluna", "")
          If IsArray(vntArray(intPos)(intCol, intRow)) Then
            vntTeste(0) = vntArray(intPos)(intCol, intRow)
            NodeColuna.Text = MontaXML(vntTeste, 1)
          Else
            NodeColuna.Text = vntArray(intPos)(intCol, intRow)
          End If
          Set NodeColuna = NodeLinha.appendChild(NodeColuna)
        Next
        Set NodeLinha = NodeTipo.appendChild(NodeLinha)
      Next
    End If
    If Nivel = 0 Then
      Set NodeTipo = NodeParametro.appendChild(NodeTipo)
      Set NodeParametro = NodePrincipal.appendChild(NodeParametro)
    End If
  Next

  If Nivel = 0 Then
    Set NodePrincipal = objDoc.appendChild(NodePrincipal)
  Else
    Set NodeTipo = objDoc.appendChild(NodeTipo)
  End If

  MontaXML = Replace(Replace(objDoc.xml, "&gt;", ">"), "&lt;", "<")

End Function

Function MontaArray(objDoc, Nivel)
  Dim arResult, arVariant, intLinhas, intColunas, intCol, intRow, i, objDoc2
  
  ReDim arResult(objDoc.documentElement.childNodes.length - 1)
  For i = 0 To objDoc.documentElement.childNodes.length - 1
    If objDoc.documentElement.childNodes(i).childNodes(0).nodeName = "Variant" Then
      intLinhas = objDoc.documentElement.childNodes(i).childNodes(0).childNodes.length
      intColunas = objDoc.documentElement.childNodes(i).childNodes(0).childNodes(0).childNodes.length
      ReDim arVariant(intColunas - 1, intLinhas - 1)
      For intRow = 0 To intLinhas - 1
        For intCol = 0 To intColunas - 1
          If objDoc.documentElement.childNodes(i).childNodes(0).childNodes(intRow).childNodes(intCol).childNodes.length <> 0 Then
            If objDoc.documentElement.childNodes(i).childNodes(0).childNodes(intRow).childNodes(intCol).childNodes(0).nodeName = "Variant" Then
              Set objDoc2 = Server.CreateObject("MSXML2.DOMDocument")
              objDoc2.loadXML ("<root><parametro>" & objDoc.documentElement.childNodes(i).childNodes(0).childNodes(intRow).childNodes(intCol).childNodes(0).xml & "</parametro></root>")
              arVariant(intCol, intRow) = MontaArray(objDoc2, 1)
            Else
              if objDoc.documentElement.childNodes(i).childNodes(0).childNodes(intRow).childNodes(intCol).Text = "True" Then
                arVariant(intCol, intRow) = True
              else
                arVariant(intCol, intRow) = objDoc.documentElement.childNodes(i).childNodes(0).childNodes(intRow).childNodes(intCol).Text
              end if
            End If
          Else
            if objDoc.documentElement.childNodes(i).childNodes(0).childNodes(intRow).childNodes(intCol).Text = "True" Then
              arVariant(intCol, intRow) = True
            else
              arVariant(intCol, intRow) = objDoc.documentElement.childNodes(i).childNodes(0).childNodes(intRow).childNodes(intCol).Text
            end if
          End If
        Next
      Next
      If Nivel = 0 Then
        arResult(i) = arVariant
      Else
        arResult = arVariant
      End If
    Else
      arResult(i) = objDoc.documentElement.childNodes(i).childNodes(0).Text
    End If
  Next
  MontaArray = arResult

End Function

Function FormataNumero(Valor, Digitos)
  Dim Ret, i
  for i = 1 to Digitos - len(valor)
    Ret = "0" & Ret
  next
  FormataNumero = Ret & Valor
End Function 

Function FormataNum( Num, Dec )
  Dim iNum, i, lFrac, iNPs
 
  If not isnumeric(Num) or not isnumeric(Dec) Then
    FormataNum = null
    Exit Function
  End If
  iNum  = int(Num)
  If Dec <> 0 Then
    lFrac =  cdbl( cdbl(Num) - int(Num) )
    If lFrac <> 0 Then
      lFrac = lFrac & "000000000000000000000000000000000"
    Else
      lFrac = "0,00000000000000000000000000000000"
    End If
    lFrac = right( lFrac , len( lFrac )-1 )
    lFrac = left( lFrac , Dec + 1)
  Else
    lFrac = ""
  End If
  For i=1 To Len(iNum) / 3
    iNPs = "." & Right(iNum , 3) & iNPs
    iNum = Left(iNum , Len(iNum)-3 )
  next
  If len(iNum) = 0 Then
    iNPs = Right(iNPs,Len(iNPs)-1)
  Else
    iNPs = iNum & iNPs 
  End If
  FormataNum = iNPs & lFrac 
end function

Function Inicializa()
  Dim Teste
  Teste = Comunica("", "")
End Function


Function Comunica(xmldoc, strOpcao)
  Dim xmlhttp
  Set xmlhttp = Server.CreateObject("Msxml2.XMLHTTP")
  xmlhttp.Open "POST", "processa.asp?Opcao=" + strOpcao, false
  xmlhttp.Send xmldoc
  Comunica = xmlhttp.responseText
End Function

Function RegistrarErro(Mensagem)
   Dim ArqPronto, ArqAcesso, ArqLog, Caminho
   Const ForReading = 1, ForWriting = 2
   
   ArqPronto = false
   
   Caminho = Request.ServerVariables("PATH_TRANSLATED")
   do while Right(Caminho, 1) <> "\"
     Caminho = Left(Caminho, len(Caminho) - 1)
   loop
   
   Set ArqAcesso = Server.CreateObject("Scripting.FileSystemObject")
   
   On Error Resume Next
   
   Err.Clear
   
   Set ArqLog = ArqAcesso.OpenTextFile(Caminho & "\Log\Log.txt", 8 , True)
   
   if Err.number <> 0 then
     Err.Clear
     Set ArqLog = ArqAcesso.CreateTextFile(Caminho & "\Log\Log.txt", False)
     if Err.number = 0 then
       ArqPronto = true
     end if
   else
     ArqPronto = true
   end if
 
   if ArqPronto then
     ArqLog.WriteLine Now & " - " & Session("Login") & " - " & Request.ServerVariables("REMOTE_HOST") & " - " & Application("Id_AplSist") & "|" & Session("Id_Modulo") & "|" & Session("Id_SubModulo") & " - " & Mensagem
     ArqLog.close
   end if
   
   Set ArqLog = Nothing
   Set ArqAcesso = nothing
End Function

%>